### ISSUE TITLE
Finish CAE implementation work

### ISSUE THREAD

**Issue by baywet** [microsoft/kiota-authentication-azure-go#13]
The current continuous access evaluation implementation returns an error when presented with a challenge, and will never be presented with a challenge to begin with. This is because of gaps in the azure identity library at the moment.
https://github.com/microsoft/kiota/issues/1179

TODO:

- [ ] update azure core reference when dependencies are unblocked
- [ ] update https://github.com/microsoft/kiota-authentication-azure-go/blob/765fdf15e7094d997e70cf356deee2d82f26569d/azure_identity_access_token_provider.go#L79
- [ ] end to end testing
- [ ] release

#12

**Comment by sebastienlevert** [comment]
@baywet am I right to assume this is completed?

**Comment by baywet** [comment]
no, it looks like we're unblocked from Azure Identity's side, but we've never completed the work here.
https://github.com/Azure/azure-sdk-for-go/issues/14931

**Comment by ikonst** [comment]
In #141 (released in v1.1.0) we've enabled CAE, but this code is still in place, so v1.1.0 appears to be a breaking change for tenants where CAE is enabled:
https://github.com/microsoft/kiota-authentication-azure-go/blob/84d8a4a8ad7c3b381c181485d0b1faa0a1bd45c6/azure_identity_access_token_provider.go#L101-L114

**Comment by baywet** [comment]
@ikonst what would you change it to? what's the issue it's causing?

**Comment by ikonst** [comment]
@baywet Do you know of any way that CAE could be triggered for a multi-tenant app? Following the docs for Entra Workflow ID, it seems only to apply on single-tenant apps.

**Comment by baywet** [comment]
I think this is because ultimately the feature is tied to a specific license, but I'm not sure. And there's no way to guarantee a specific tenant will have any given license in a multi-tenant context. Again, don't quote me on that, my confidence is barely 20%
